### DISCUSSION TITLE
Dify Apache 2.0 License

### DISCUSSION THREAD

**Discussion by abhishekbpandit** [langgenius/dify#10139] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

I read the dify license and saw that it is licensed under Apache 2.0 but with additional restrictions. I searched online and all sources tell me that one can't put additional restrictions on Apache 2.0. The restrictions added by dify are contrary to Apache 2.0's agreement which allows unrestricted use and distribution of the code. 

Does anybody have thoughts on this?

**Comment by nnWhisperer** [comment]
https://github.com/langgenius/dify/issues/14336
